File: DevExpress-Demo/CS/PropertyGridDemo.Wpf/DemoTesting/Fixtures.cs
using DevExpress.Xpf.DemoBase.DemoTesting;
using System.Threading;
using System.Windows.Threading;
using System;
using DevExpress.Xpf.Core.Native;
using DevExpress.Xpf.Editors;
using DevExpress.Xpf.Editors.Helpers;
using System.Windows;
using System.Globalization;
using DevExpress.Xpf.Core;
using DevExpress.Xpf.DemoBase;
using System.Linq;

namespace PropertyGridDemo.Tests {
    public class PropertyGridCheckAllDemosFixture : CheckAllDemosFixture {
    }
    public class PropertyGridDemoModulesAccessor : DemoModulesAccessor<PropertyGridDemoModule> {
        public PropertyGridDemoModulesAccessor(BaseDemoTestingFixture fixture)
            : base(fixture) {
        }
    }
    public abstract class BasePropertyGridDemoTestingFixture : BaseDemoTestingFixture {
        protected PropertyGridDemoModulesAccessor ModuleAccessor { get; private set; }
        public BasePropertyGridDemoTestingFixture() {
            ModuleAccessor = GetModulesAccessor();
        }
        protected abstract PropertyGridDemoModulesAccessor GetModulesAccessor();
    }
}

File: DevExpress-Demo/VB/PropertyGridDemo.Wpf/DemoTesting/Fixtures.vb
Imports DevExpress.Xpf.DemoBase.DemoTesting
Imports DevExpress.Xpf.Editors.Helpers

Namespace PropertyGridDemo.Tests

    Public Class PropertyGridCheckAllDemosFixture
        Inherits CheckAllDemosFixture

    End Class

    Public Class PropertyGridDemoModulesAccessor
        Inherits DemoModulesAccessor(Of PropertyGridDemoModule)

        Public Sub New(ByVal fixture As BaseDemoTestingFixture)
            MyBase.New(fixture)
        End Sub
    End Class

    Public MustInherit Class BasePropertyGridDemoTestingFixture
        Inherits BaseDemoTestingFixture

        Private _ModuleAccessor As PropertyGridDemoModulesAccessor

        Protected Property ModuleAccessor As PropertyGridDemoModulesAccessor
            Get
                Return _ModuleAccessor
            End Get

            Private Set(ByVal value As PropertyGridDemoModulesAccessor)
                _ModuleAccessor = value
            End Set
        End Property

        Public Sub New()
            ModuleAccessor = GetModulesAccessor()
        End Sub

        Protected MustOverride Function GetModulesAccessor() As PropertyGridDemoModulesAccessor
    End Class
End Namespace
